### DISCUSSION TITLE
How to retrieve a WidgetRef when just the BuildContext  is available?

### DISCUSSION THREAD

**Discussion by jeduden** [rrousselGit/riverpod#728] [Q&A] [answered]
I am looking at the riverpod example in https://github.com/slovnicki/beamer/blob/master/examples/authentication_riverpod/lib/main.dart#L23
which is depends on `flutter_riverpod: ^0.14.0+3`:

```
check: (context, state) =>
              context.read(authProvider).status == AuthStatus.authenticated,
```
since `context.read` is not available anymore - how would you suggest to retrieve `WidgetRef` in this case ? is a change in Beamer needed?

**Comment by rrousselGit** [comment, marked as answer]
You can't get a ref from just a context.

You would need a Consumer for this.

**Reply by jeduden** [reply]
slovnicki will add ideas to solve this problem to https://github.com/slovnicki/beamer/issues/369

**Reply by julgog** [reply]
hello @rrousselGit , I see that you have a lot of activity and I admire you for providing us with excellent libraries, I am currently developing my entire project with riverpod.

I have a question about the use of go_router with riverpod and firebase, I have not been able to find a simple, functional and nice example which I can integrate.

I have read that refreshlistenable will be obsolete, so I have seen an example with inheritnotifier and detect changes in redirect with the context as indicated in this example fragment:
<code>
     redirect: (BuildContext context, GoRouterState state) async {
       // Using `of` method creates a dependency of StreamAuthScope. It will
       // cause go_router to repair current route if StreamAuth has new sign-in
       // information.
       final bool loggedIn = await StreamAuthScope.of(context).isSignedIn();
.....
</code>
I can't make an equivalent in riverpod, so I thought of doing this.
is my example correct?
<code>
@override
Widget build(BuildContext context, WidgetRef ref) {
log("inside main build root");
final _router= GoRouter(
   initialLocation: SplashPage.routeLocation,
   routes: [
      ...
   ],
   redirect: (context, state) {
     log("inside redirection");
     var authState = ref.read(authProvider);
     .....
   }
);
ref.listen(authProvider, (previoustValue, newValue) {
log("change value");
_router.refresh();
});
return MaterialApp.router(
routerConfig: _router,
title: "My example Go_router - Riverpod - Firebase",
);
}
</code>
